storage/replica.go

```diff
@@ -1484,9 +1485,21 @@ func (r *Replica) resolveIntents(ctx context.Context, intents []proto.Intent) {
 				if log.V(1) {
 					log.Warningc(ctx, "batch resolve failed: %s", err)
 				}
-				if _, ok := err.(*proto.RangeKeyMismatchError); !ok {
-					// TODO(tschottdorf)
-					panic(fmt.Sprintf("intent resolution failed, error: %s", err.Error()))
+				// At this point, as long as the local Replica accepts the
+				// request it should never fail. However, the replica may reject
+				// the request in certain cases (for example, if the replica has
+				// been removed from its range via a rebalancing a command).
+				// Therefore, we inspect the returned error to detect cases
+				// where the command was rejected, and can safely ignore those
+				// errors.
+				if err != multiraft.ErrGroupDeleted {
+					switch err.(type) {
+					case *proto.RangeKeyMismatchError:
+					case *proto.NotLeaderError:
+					default:
+						// TODO(tschottdorf): Does this need to be a panic?
+						panic(fmt.Sprintf("intent resolution failed with unexpected error: %s", err.Error()))
```

Comment:
`Error()` has higher precedence than `String()` in `Sprintf`'s implicit conversions (search for "if an operand implements the error interface" in https://golang.org/pkg/fmt/), so I believe `Sprintf("%s", err)` and `Sprintf("%s", err.Error())` are exactly equivalent (including the number of allocations) in all cases. I prefer to not use `.Error()` just because it's unnecessary, but there is no functional difference.